src/index/index.h

```diff
@@ -28,82 +29,97 @@ typedef struct {
-	IndexType type      // Index is a fulltext index
-);
-
-// Adds field to index.
-void Index_AddField
-(
-	Index *idx,
-	const char *field
-);
-
-// Removes fields from index.
-void Index_RemoveField
-(
-	Index *idx,
-	const char *field
-);
-
-// Index node.
-void Index_IndexNode
-(
-	Index *idx,     // Index to use
-	const Node *n   // Node to index
-);
-
-// Remove node from index.
-void Index_RemoveNode
-(
-	Index *idx,     // Index to use
-	const Node *n   // Node to remove
-);
-
-// Constructs index.
-void Index_Construct
-(
-	Index *idx
-);
-
-// Query index.
-RSResultsIterator *Index_Query
-(
-	const Index *idx,
-	const char *query,          // Query to execute
-	char **err                  // Optional, report back error
-);
-
-// Return indexed label.
-const char *Index_GetLabel
-(
-	const Index *idx
-);
-
-// Returns number of fields indexed.
-uint Index_FieldsCount
-(
-	const Index *idx
-);
-
-// Returns indexed fields.
-const char **Index_GetFields
-(
-	const Index *idx
-);
-
-// Checks if given field is indexed.
-bool Index_ContainsField
-(
-	const Index *idx,
-	const char *field
-);
-
-// Free fulltext index.
-void Index_Free
-(
-	Index *idx
-);
+/**
+ * @brief  Create a new index.
+ * @param  *label: Indexed label
+ * @param  type: Index type - exact match or full text.
+ * @retval New constructed index for the label.
+ */
+Index *Index_New(const char *label, IndexType type);
+
+/**
+ * @brief  Adds field to index.
+ * @param  *idx: Index
+ * @param  *field: Field to add.
+ */
+void Index_AddField(Index *idx, const char *field);
+
+/**
+ * @brief  Removes attribute from index.
+ * @param  *idx: Index
+ * @param  attribute_id: Attribute to remove
+ */
+void Index_RemoveField(Index *idx, Attribute_ID attribute_id);
+
+/**
+ * @brief  Index node.
+ * @param  *idx: Index
+ * @param  *n :Node
+ */
+void Index_IndexNode(Index *idx, const Node *n);
+
+/**
+ * @brief  Remove node from index.
+ * @param  *idx: Index to remove the node from.
+ * @param  *n: Node to remove.
+ */
+void Index_RemoveNode(Index *idx, const Node *n);
+
+/**
+ * @brief  Constructs index.
+ * @param  *idx:
+ */
+void Index_Construct(Index *idx);
+
+/**
+ * @brief  Query an index.
+ * @param  *idx: Index.
+ * @param  *query: Query to execute.
+ * @param  **err: Optional, report back error
+ * @retval RedisSearch results iterator.
+ */
+RSResultsIterator *Index_Query(const Index *idx, const char *query, char **err);
+
+/**
+ * @brief Return indexed label.
+ * @param  *idx: Index.
+ * @retval Index's label.
+ */
+const char *Index_GetLabel(const Index *idx);
+
+/**
+ * @brief  Returns number of fields indexed.
+ * @param  *idx: Index.
+ * @retval Number of indexed fields.
+ */
+uint Index_FieldsCount(const Index *idx);
+
+/**
+ * @brief  Returns indexed fields.
+ * @note   Returns a shallow copy.
+ * @param  *idx: Index to extract fields from.
+ * @retval Array with the indexed fields.
+ */
+const char **Index_GetFields(const Index *idx);
```

Comment:
I guess this is comfortable, but I think the entire interaction with the index should be with numeric Ids